lib/galaxy/config/__init__.py

```diff
@@ -139,40 +97,112 @@ def resolve_path(path, root):
     return path
 
 
-def find_path(kwargs, var, root):
-    """Find a configuration path that may exist at different defaults."""
-    defaults = PATH_DEFAULTS[var]
-
-    if kwargs.get(var, None) is not None:
-        path = kwargs.get(var)
-    else:
-        for default in defaults:
-            if default is None:
-                # if None is the final default - just return that.
-                return None
-            if os.path.exists(resolve_path(default, root)):
-                path = default
-                break
-        else:
-            path = defaults[-1]
-
-    return resolve_path(path, root)
-
-
 def find_root(kwargs):
     root = kwargs.get('root_dir', '.')
     return root
 
 
-class Configuration(object):
+class BaseAppConfiguration(object):
+    def _set_config_base(self, config_kwargs):
+        self.sample_config_dir = os.path.join(os.path.dirname(__file__), 'sample')
+        self.config_file = find_config_file('galaxy')
+        # Parse global_conf and save the parser
+        self.global_conf = config_kwargs.get('global_conf', None)
+        self.global_conf_parser = configparser.ConfigParser()
+        if not self.config_file and self.global_conf and "__file__" in self.global_conf:
+            self.config_file = self.global_conf['__file__']
+        if self.config_file is None:
+            log.warning("No Galaxy config file found, running from current working directory: %s", os.getcwd())
+        else:
+            try:
+                self.global_conf_parser.read(self.config_file)
+            except (IOError, OSError):
+                raise
+            except Exception:
+                # Not an INI file
+                pass
+        self.config_dir = config_kwargs.get('config_dir', os.path.dirname(self.config_file or os.getcwd()))
```

Comment:
@natefoo `os.path.dirname(os.getcwd())` is the parent of the current directory, so I think this may be wrong if `self.config_file` is None.
Also this is set to a different value on line 134 if `running_from_source` is True.